mkosi/initrd.py

```diff
@@ -141,12 +142,22 @@ def initrd_finalize(staging_dir: str, output: str, output_dir: str) -> None:
     else:
         output_dir = os.fspath(Path.cwd())
 
-    log_notice(f"Copying {staging_dir}/{output} to {output_dir}/{output}")
+    staging = Path(f"{staging_dir}/{output}")
+    tmp = Path(f"{output_dir}/{output}.new")
+    final = Path(f"{output_dir}/{output}")
+
+    log_notice(f"Copying {staging} to {tmp}")
     # mkosi symlinks the expected output image, so dereference it
-    copy_tree(
-        Path(f"{staging_dir}/{output}").resolve(),
-        Path(f"{output_dir}/{output}"),
-    )
+    try:
+        copy_tree(staging.resolve(), tmp)
+    except subprocess.CalledProcessError:
+        rmtree(tmp)
+        raise
```

Comment:
Since the except is here only to ensure cleanup, we can move it into a `finally` instead, which is a bit more concise and saves the `import subprocess`.
```suggestion
finally:
        rmtree(tmp)
```

Reply:
Scratch that nonsense, I'll get some coffee…